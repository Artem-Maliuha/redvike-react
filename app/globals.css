@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --primary: #1A1A1E;
  --background-rgb: 255, 255, 255;
}

body {
  color: rgb(var(--primary));
  background: rgb(var(--background-rgb));
}
