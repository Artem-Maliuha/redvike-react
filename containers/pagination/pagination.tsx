import Image from 'next/image'
import { PageButton } from '@/components/page-button'
interface PaginationProps {
  baseUrl?: string
  currentPage: number
  perPageCount: number
  elementsCount: number
}
export const Pagination = (props: PaginationProps) => {
  const { baseUrl = '', currentPage, perPageCount, elementsCount } = props
  const pagesCount = Math.floor(elementsCount / perPageCount)

  const prevLink =
    currentPage < 3 ? `${baseUrl}/` : `${baseUrl}/?page=${currentPage - 1}`

  const nextLink =
    currentPage === pagesCount
      ? `${baseUrl}/?page=${pagesCount}`
      : `${baseUrl}/?page=${currentPage + 1}`

  return (
    <div className="flex flex-row items-end gap-[12px]">
      <PageButton href={prevLink}>
        <Image
          src="/assets/images/chevron.svg"
          alt="<"
          width={20}
          height={20}
        />
      </PageButton>
      <PageButton href={`${baseUrl}/`}>1</PageButton>
      <span className="flex items-end font-bold">...</span>
      <PageButton
        active
        href={
          currentPage !== 1 ? `${baseUrl}/?page=${currentPage}` : `${baseUrl}/`
        }
      >
        {currentPage}
      </PageButton>
      <span className="flex items-end font-bold">...</span>
      <PageButton href={`${baseUrl}/?page=${pagesCount}`}>
        {pagesCount}
      </PageButton>
      <PageButton href={nextLink}>
        <Image
          src="/assets/images/chevron.svg"
          alt=">"
          width={20}
          height={20}
          className="rotate-180"
        />
      </PageButton>
    </div>
  )
}
