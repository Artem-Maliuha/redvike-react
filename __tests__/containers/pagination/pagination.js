import renderer from 'react-test-renderer'
import { describe, it, expect } from '@jest/globals'
import { Pagination } from '../../../containers/pagination'

describe('<Pagination />', () => {
  it('snapshot', () => {
    const component = renderer.create(
      <Pagination currentPage={1} elementsCount={100} perPageCount={10} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('component length', () => {
    const component = renderer.create(
      <Pagination currentPage={1} elementsCount={100} perPageCount={10} />
    )
    let tree = component.toJSON()
    expect(tree.children.length).toEqual(7)
  })
})
