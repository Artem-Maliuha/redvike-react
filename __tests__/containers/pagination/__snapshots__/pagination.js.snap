// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> snapshot 1`] = `
<div
  className="flex flex-row items-end gap-[12px]"
>
  <a
    className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primaryfalse"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <img
      alt="<"
      data-nimg="1"
      decoding="async"
      height={20}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/assets/images/chevron.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={20}
    />
  </a>
  <a
    className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primaryfalse"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    1
  </a>
  <span
    className="flex items-end font-bold"
  >
    ...
  </span>
  <a
    className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primary bg-primary text-white"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    1
  </a>
  <span
    className="flex items-end font-bold"
  >
    ...
  </span>
  <a
    className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primaryfalse"
    href="/?page=10"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    10
  </a>
  <a
    className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primaryfalse"
    href="/?page=2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <img
      alt=">"
      className="rotate-180"
      data-nimg="1"
      decoding="async"
      height={20}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/assets/images/chevron.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={20}
    />
  </a>
</div>
`;
